ci/README.md

```diff
@@ -48,12 +48,55 @@ following:
 
 ## `check_enterprise_merge.sh`
 
-When you open a PR on community, if it creates a conflict with
-enterprise-master, the check-merge-to-enterprise will fail. Say your branch name
-is `$PR_BRANCH`, we will refer to `$PR_BRANCH` on community as
-`community/$PR_BRANCH` and on enterprise as `enterprise/$PR_BRANCH`. If the
-job already passes, you are done, nothing further required! Otherwise follow the
-below steps. First make sure these two things are the case:
+This check exists to make sure that we can always merge the `master` branch of
+`community` into the `enterprise-master` branch of the `enterprise` repo.
+There are two conditions in which this check passes:
+1. There are no merge conflicts between your PR branch and `enterprise-master`.
+2. There are merge conflicts, but there is a branch with the same name in the
+   enterprise repo that:
+   1. Contains the last commit of the community branch with the same name.
+   2. Merges cleanly into `enterprise-master`
+
+If the job already passes, you are done, nothing further required! Otherwise
+follow the below steps.
+
+### Prerequisites
+
+Before continuing with the real steps make sure you have done the following
+(this only needs to be done once):
+1. You have enabled `git rerere` ([docs](https://git-scm.com/docs/git-rerere),
```

Comment:
```suggestion
1. You have enabled `git rerere` ([docs](https://git-scm.com/docs/git-rerere) in your enterprise repo,
```

Reply:
`git config --global rerere.enabled true` command enables rerere for all repos.

Reply:
ops, that's true

Reply:
but I guess, enabling rerere with no global option and only for enterprise repo might be safer I guess ?

Reply:
I added both options, so someone can choose.

Reply:
Remember when we had merge conflicts on community/release-9.x branches, and again on enterprise/release-9.x. I think it is better to enable it globally.